cmd/juju/application/trust.go

```diff
@@ -62,3 +75,19 @@ func (c *trustCommand) Init(args []string) error {
 	trustOptionPair = fmt.Sprintf("%s=%t", application.TrustConfigOptionName, !c.removeTrust)
 	return c.parseSet([]string{trustOptionPair})
 }
+
+func (c *trustCommand) Run(ctx *cmd.Context) error {
+	modelType, err:= c.ModelType()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if modelType == model.CAAS && !c.removeTrust {
```

Comment:
This should really be validated on the API side and not in the CLI, shouldn't it?
Checking ModelType in the CLI feels like an anti pattern.
I guess we can't really since we aren't actually calling an API around trust, we are just setting trust: true in config.

Reply:
The model info (UUID and type) is stored client side after making an API call to get it if the current client doesn't know about the model yet.